server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -734,6 +734,23 @@ public static IndexMergePolicy fromString(String text) {
         Property.IndexScope
     );
 
+    public static final Setting<Long> INDEX_CONTEXT_CREATED_VERSION = Setting.longSetting(
+        "index.context.created_version",
+        0,
+        0,
+        Property.PrivateIndex,
+        Property.IndexScope
+    );
+
+    public static final Setting<Long> INDEX_CONTEXT_CURRENT_VERSION = Setting.longSetting(
+        "index.context.current_version",
+        0,
+        0,
+        Property.PrivateIndex,
+        Property.Dynamic,
```

Comment:
shouldn't this be controlled by the system and not allowed for user to change?

Reply:
Since this is a private index setting, it should not be modified by user.
I've removed dynamic for now. If needed, will add if we decide to have flows around updating context on created indices.